src/Discord.Net.Webhook/DiscordWebhookClient.cs

```diff
@@ -43,6 +50,41 @@ public DiscordWebhookClient(IWebhook webhook, DiscordRestConfig config)
             _webhookId = Webhook.Id;
         }
 
+        /// <summary>
+        ///     Creates a new Webhook Discord client.
+        /// </summary>
+        /// <param name="webhookUrl">The url of the webhook.</param>
+        /// <param name="config">The configuration options to use for this client.</param>
+        /// <exception cref="ArgumentException">Thrown if the <paramref name="webhookUrl"/> is an invalid format.</exception>
+        /// <exception cref="ArgumentNullException">Thrown if the <paramref name="webhookUrl"/> is null or whitespace.</exception>
+        public DiscordWebhookClient(string webhookUrl, DiscordRestConfig config) : this(config)
+        {
+            if (string.IsNullOrWhiteSpace(webhookUrl))
+                throw new ArgumentNullException(paramName: nameof(webhookUrl), message:
+                    "The given webhook Url cannot be null or whitespace.");
+
+            // thrown when groups are not populated/valid, or when there is no match
+            ArgumentException ex()
```

Comment:
Add an optional argument to this location function to specify whether the id or the token was what didn't pass the format check.